File: algorithms/[168]Excel表列名称/solution.py
# topics = ["数学"]


class Solution:
    def convertToTitle(self, columnNumber: int) -> str:
        """
        Math
        time O(log26columnNumber), space O(log26columnNumber)
        """
        res = ''
        n = 26
        while columnNumber:
            d = columnNumber % n
            if d == 0:
                d = n
                columnNumber -= n
            res = chr(d + 64) + res
            columnNumber //= n

        return res

File: algorithms/[168]Excel表列名称/solution.ts
// topics = ["数学"]

/**
 * Math
 * time O(log26columnNumber), space O(log26columnNumber)
 */
function convertToTitle(columnNumber: number): string {
  const n = 26;
  let res = '';

  while (columnNumber) {
    let d = columnNumber % n;
    if (d == 0) {
      d = n;
      columnNumber -= n;
    }
    res = String.fromCharCode(d + 64) + res;
    columnNumber = Math.floor(columnNumber / n);
  }

  return res;
}
